app/interceptors/MetricsInterceptor.java

```diff
@@ -0,0 +1,50 @@
+package interceptors;
+
+import static org.apache.http.HttpHeaders.USER_AGENT;
+import static org.sagebionetworks.bridge.BridgeConstants.METRICS_EXPIRE_SECONDS;
+import static org.sagebionetworks.bridge.BridgeConstants.X_FORWARDED_FOR_HEADER;
+import models.Metrics;
+import models.RequestUtils;
+
+import org.aopalliance.intercept.MethodInterceptor;
+import org.aopalliance.intercept.MethodInvocation;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.stereotype.Component;
+
+import play.cache.Cache;
+import play.mvc.Http;
+import play.mvc.Http.Request;
+import play.mvc.Result;
+
+@Component("metricsInterceptor")
+public class MetricsInterceptor implements MethodInterceptor {
+
+    private static final Logger logger = LoggerFactory.getLogger(MetricsInterceptor.class);
+
+    @Override
+    public Object invoke(MethodInvocation method) throws Throwable {
+        final Metrics metrics = initMetrics();
+        Cache.set(metrics.getCacheKey(), metrics, METRICS_EXPIRE_SECONDS);
+        try {
+            final Result result = (Result)method.proceed();
+            metrics.setStatus(result.toScala().header().status());
+            return result;
+        } finally {
+            Cache.remove(metrics.getCacheKey());
+            metrics.end();
+            logger.info(metrics.toJsonString());
```

Comment:
Is there any other structured format that would be easier to search? I think JSON is okay but it might be messy to read in the logs and it's a little tedious to search for stuffy like "requestId":"foo" if requestId=foo would work.